docs/environment-and-software/vscode.md

```diff
@@ -0,0 +1,197 @@
+
+![Explorer Icon](vscode/media/explorer-view-icon.png)
+
+The Explorer tab allows a graphical file structure for the path that you provide.  If you are already connected to a Derecho or Casper login node then you will see the GLADE file structure populated.
+
+!!! warning 
+    VS Code will be able to view all subdirectories of your path when connecting the Explorer. Be mindful that a higher total number of files in the Explorer space will increase your memory and CPU utilization on the login nodes by default.  We recommend only selecting the path for your current project and the files that you need to directly interact with during that session.  Instead of connecting to `/glade/work/$USER`, connect to your project folder of `/glade/work/$USER/myproject/code`.  Smaller Explorer spaces also help with performance for addons that use code completion like Intellisense.
+
+!!! tip
+    Reduce the number of files that VS Code scans periodically to lower CPU utilization, memory, and Intellisense scanning time on the login nodes.  Add these paths to your Settings file found in *Code > Preferences > Settings*, search for the strings below, and add the wildcard items to each file exclude type:
+
+    ```
+    "files.watcherExclude": {
+        "**/.git/objects/**": true,
+        "**/.git/subtree-cache/**": true,
+        "**/node_modules/**": true,
+        "**/dist/**": true,
+        "**/conda-envs/**": true
+        "**/env/**": true,
+    },
+    "files.exclude": {
+        "**/.git/objects/**": true,
+        "**/node_modules/**": true,
+    },
+    "search.exclude": {
+        "**/node_modules/**": true,
+        "**/dist/**": true,
+        "**/tmp/**": true,
+        "**/.git/objects/**": true,
+        "**/.git/subtree-cache/**": true,
+        "**/.DS_Store": true,
+        "**/.vscode": true,
+        "**/__pycache__": true,
+        "**/.pytest_cache": true,
+    },
+    ```
+
+### Source Control
+
+VS Code has integrated source control management (SCM) and includes Git support natively. Access the Source Control view by clicking on the Source Control icon in the Activity Bar on the side of VS Code or the *View: Source Control* command (⇧⌘G)
+
+![Source Control](vscode/media/sourcecontrol-view-icon.png)
+
+### Extensions
+
+Browse and install extensions from within VS Code. Bring up the Extensions view by clicking on the Extensions icon in the Activity Bar on the side of VS Code or the *View: Extensions* command (⇧⌘X).
+
+![Extension Icon](vscode/media/extensions-view-icon.png)
+
+Search for extensions within the Extensions tab and click the extension to install, disable, or view details about it.  The `...` button will provide additional options to enable or disable all extensions, update current extensions, or view all running extensions.
+
+![Extension Details](vscode/media/extensions-list.png)
+
+!!! info 
```

Comment:
I would suggest changing this from "info" to "warning" or "danger". But thanks for including this.

Reply:
This has been changed to danger.

Reply:
Awesome. Thanks!